### PR TITLE
Fixed up accuracy layer.

### PR DESCRIPTION
**Pull request by BlGene**
This PR extends the accuracy layer to behave more like the SoftmaxWithLoss layer. It allows bottom datum to have a width and height, as well as the ignore_label option. Original functionality of the layer remains unchanged.

### CODE REVIEW

File: src/caffe/proto/caffe.proto [hunk 1]
```diff
@@ -30,6 +30,10 @@ message Datum {
   optional bool encoded = 7 [default = false];
 }
 
+message DatumVector {
```
Comment (BlGene) [inline src/caffe/proto/caffe.proto]: This is added because it allows one to define databases that store several Datums to one keyword.

**Comment by BlGene** [comment]
The PR timed out, it AFAIK it builds and tests fine.

File: src/caffe/proto/caffe.proto (hunk 1 above)
Comment (sguada) [inline src/caffe/proto/caffe.proto]: Leave that for another PR, where it is used.

Comment (shelhamer) [inline src/caffe/proto/caffe.proto]: Seconding @sguada. Note also that there is a BlobProtoVector that could be
used for packing multiple pieces of data too.

On Sun, Feb 22, 2015 at 4:01 PM, Sergio Guadarrama <notifications@github.com

**Comment by BlGene** [comment]
@sguada @shelhamer I concede DatumVector is out.

**Comment by philkr** [comment]
Nice PR. I'm currently using it and it works as advertised.
One small remark: As this now operates on larger blobs it might make sense to also have a GPU implementation similar to SoftmaxWithLoss.

**Comment by shelhamer** [comment]
Thanks for the accuracy layer extension. This needs a rebase after the generalization to N-D blobs in #1970.

**Comment by BlGene** [comment]
@shelhamer @jeffdonahue 
Hi Guys, I updated the PR to include the changes from #2076.

@philkr Thanks. I don't have a GPU implementation planned ATM, but I will push one as soon as I have one.

**Comment by jeffdonahue** [comment]
Thanks @BlGene, in general please rebase your code rather than merging.  I squashed and cherry-picked your changes in #2076.

**Comment by wkal** [comment]
Could you provide a little demo to show how to take advantage the new feature of this PR? From the beginning description, I basically know the function of this PR, however while I review the source code in the caffe/src, and I found that the layers still in old faction (num, channels, width, height), not see any changes that related to this PR (support N-Dimension data of convolution and pooling).. Thanks!
